src/line_buffer.rs

```diff
@@ -135,16 +141,28 @@ impl LineBuffer {
         self.insertion_point.offset
     }
 
+    pub fn insert_char(&mut self, c: char) {
+        let pos = self.insertion_point();
+        self.lines[pos.line].insert(pos.offset, c);
+        self.move_right();
+    }
+
     /// Insert a single character at the given cursor postion
-    pub fn insert_char(&mut self, pos: InsertionPoint, c: char) {
+    pub fn insert_char_at(&mut self, pos: InsertionPoint, c: char) {
         self.lines[pos.line].insert(pos.offset, c)
     }
```

Comment:
I believe we don't really have a use-case where the newly created `insert_char` is not sufficient, we can remove this method from the public interface and only use that.

Reasoning: 
It leads to an inconsistent internal state and the user has to be cognizant about making sure that the insertion point is updated properly. I think we should only expose this sort of API if there is something we can't do with the high-level API that we have.

Reply:
Right now there is not really any place we are making use of this internally so, I am removing this for now. If there is a use for this we can re-add this. Is that ok?